Refactoring: file structure

rubenthoms opened the pull request:
Refactored file structure in order to have it more similar to webviz-subsurface-components.

asnyv commented:
Suggest that you separate out the refactoring to another PR that we can merge first. Makes it easier to track changes.

Also: Do you think it is possible in some way to move the subsurface specific code somewhere else? E.g. the icons for Field, Well and etc?

HansKallekleiv commented on .gitignore:
```diff
@@ -17,7 +17,6 @@ inst/
 .vscode/*
 !.vscode/settings.json
 !.vscode/tasks.json
-!.vscode/launch.json
```
Just curious - why remove this line?

rubenthoms replied:
@HansKallekleiv  As I understand .gitignore, all matching the patterns in there is not committed. However, using ! as a negator, those lines are exceptions for the afore written patterns. This means that we would not commit anything inside .vscode/ except for settings.json, tasks.json and launch.json. Do we want to commit userspecific VS Code settings? I was assuming not, but in that case we should also remove the other exceptions. What do you think, @asnyv?

asnyv replied:
Seems like it is quite a common thing (and not something we have added manually): https://stackoverflow.com/a/32979933
Mainly a question of preferences I guess: should user settings, debug setup and etc be shared among us. 

I am happy with removing all of them, but also ok with keeping them as is.

rubenthoms replied:
I put it back in again since we also keep the other ones. We can decide another time and remove it in another PR if we conclude that we don't need it.

HansKallekleiv reviewed (approved):
LGTM 👍 just one minor question

HansKallekleiv commented:
> Suggest that you separate out the refactoring to another PR that we can merge first. Makes it easier to track changes.
> 
> Also: Do you think it is possible in some way to move the subsurface specific code somewhere else? E.g. the icons for Field, Well and etc? This repository should be clean of anything non-generic.

Yes, good point. Ideally this should perhaps be hosted somewhere as we do for the logo in the [equinor theme](https://github.com/equinor/webviz-config-equinor/blob/7a1ce568c820f78090db506e24a86ed3b0889089/webviz_config_equinor/assets/equinor_theme.css#L144)

I'm not sure if there are any existing solution for serving subsurface related "assets". 
Could this be within sumo's domain similar to how the schemas are served? @perolavsvendsen

perolavsvendsen commented:
> I'm not sure if there are any existing solution for serving subsurface related "assets".
> Could this be within sumo's domain similar to how the schemas are served? @perolavsvendsen

Meh... We serve the schemas as a simple dedicated webapp. Hope to not have to do this in the future ("someone" should do that). When it comes to visual components, no such needs nor plans. I understood this to be icons.

asnyv commented:
Think this PR has changed into a pure refactor PR since my first comment @HansKallekleiv 😊 so nothing subsurface left in this one. But the MPLv2 headers from when it was in webviz-subsurface-components have to be removed before this can be merged (as this repo is MIT licensed).

asnyv reviewed (changes requested):
Adding the change request to make sure we don't accidentally merge before we remove MPLv2 headers 😊 (insert copyright headers instead)

rubenthoms commented:
> Adding the change request to make sure we don't accidentally merge before we remove MPLv2 headers 😊 (insert copyright headers instead)

I removed all MPL notices and added MIT notices instead.

asnyv commented:
> > Adding the change request to make sure we don't accidentally merge before we remove MPLv2 headers 😊 (insert copyright headers instead)
> 
> @asnyv I removed all MPL notices and added MIT notices instead.

Nice 😊 Should we move this PR out of draft now @rubenthoms?

asnyv reviewed (approved):
LGTM 👍
